runtime/compiler/control/J9Options.cpp

```diff
@@ -1989,11 +1989,23 @@ J9::Options::fePreProcess(void * base)
          std::random_device rd;
          std::mt19937_64 rng(rd());
          std::uniform_int_distribution<uint64_t> dist;
-         compInfo->getPersistentInfo()->setClientUID(dist(rng));
+         // We want to generate a non-zero client UID for client mode
+         uint64_t clientUID = dist(rng);
+         while (0 == clientUID)
+            clientUID = dist(rng);
+         compInfo->getPersistentInfo()->setClientUID(clientUID);
+         jitConfig->clientUID = clientUID;
+
          // _safeReservePhysicalMemoryValue is set as 0 for the JITClient because compilations
          // are done remotely. The user can still override it with a command line option
          J9::Options::_safeReservePhysicalMemoryValue = 0;
          }
+      else
+         {
+         // Client UID is zero in jitserver mode
```

Comment:
We will need something similar for a JVM that has JITServer code compiled-in, but it's acting like a regular JVM (not server, not client)

Reply:
Shall we set clientUID to 1 if running as a regular JVM?

Reply:
Why not clientUIT==0 for regular JVM and for server. Some value just for client.